### ISSUE TITLE
[no-invalid-this] False positive when using this to declare an attribute

### ISSUE THREAD

**Issue by flowstate247** [typescript-eslint/typescript-eslint#2324]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-invalid-this": ["warn"]
  }
}
```

```TS
class Example1 {
  i = 0
  inc(): void { this.i++ }
  alias = this.inc // Warning here
}

class Example2 {
  i = 0
  constructor() { this.alias = this.inc	}
  inc(): void { this.i++ }
  alias: () => void
}
```

**Expected Result**
No warning shown

**Actual Result**
Show warning "Unexpected 'this'" in the Example1 class

**Additional Info**
Example1 and Example2 classes are both valid TypeScript and generate identical JavaScript code.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.7.0` |
| `@typescript-eslint/parser`        | `3.7.0` |
| `TypeScript`                       | `3.9.7` |
| `ESLint`                           | `7.5.0` |
| `node`                             | `v14.1.0` |
| `npm`                              | `6.14.4` |

**Comment by bradzacher** [comment]
#491
